app/src/main/java/com/google/android/material/bottomsheet/ViewPagerBottomSheetBehavior.java

```diff
@@ -31,8 +32,8 @@ View findScrollingChild(View view) {
             return view;
         }
 
-        if (view instanceof ViewPager) {
-            ViewPager viewPager = (ViewPager) view;
+        if (view instanceof ViewPager2) {
+            ViewPager2 viewPager = (ViewPager2) view;
             View currentViewPagerChild = viewPager.getChildAt(viewPager.getCurrentItem());
```

Comment:
> With your changes, it is no longer possible to scroll up in the chapters list and the shownotes list (because every time when scrolling up, the slide-up sheet is closed).

I did understand the reasoning behind the code existence and have tested it locally, however I don't see a difference before these changes, assuming it is working as expected as up scrolling is still possible if one first goes for down scrolling then up scrolling. I will investigate more but can you check also if this really worked before my changes also?

> I am not really happy about getChildAt(0), so maybe you have another idea about that, too (IDs?).

Yes, that's not reliable. I used setBackground(Color.RED) to see if I've found the pages correctly.

Reply:
> assuming it is working as expected as up scrolling is still possible if one first goes for down scrolling then up scrolling

Using the old version, I do not need to scroll up before being able to scroll down. I can just directly scroll up. In your version, I am also unable to hide the audio player by scrolling on the cover page.

> I used setBackground(Color.RED) to see if I've found the pages correctly.

When printing `currentViewPagerChild` in your current version, you can see that it always finds the RecyclerView that the ViewPager2 is based on. Not the Views within the pager.

Reply:
Oh, I am sorry, you already had provided the fix 🤦🤦‍♂️ https://github.com/AntennaPod/AntennaPod/pull/4020/commits/fca01a420ff267038a4d4b44e882cbf5c0b90667